src/spider/scheduler/scheduler.cpp

```diff
@@ -137,30 +166,10 @@ auto main(int argc, char** argv) -> int {
     spdlog::set_level(spdlog::level::trace);
 #endif
 
-    boost::program_options::variables_map const args = parse_args(argc, argv);
-
     unsigned short port = 0;
     std::string scheduler_addr;
     std::string storage_url;
-    try {
-        if (!args.contains("port")) {
-            spdlog::error("port is required");
-            return cCmdArgParseErr;
-        }
-        port = args["port"].as<unsigned short>();
-        if (!args.contains("host")) {
-            spdlog::error("host is required");
-            return cCmdArgParseErr;
-        }
-        scheduler_addr = args["host"].as<std::string>();
-        if (!args.contains("storage_url")) {
-            spdlog::error("storage_url is required");
-            return cCmdArgParseErr;
-        }
-        storage_url = args["storage_url"].as<std::string>();
-    } catch (boost::bad_any_cast& e) {
-        return cCmdArgParseErr;
-    } catch (boost::program_options::error& e) {
+    if (!parse_args(argc, argv, scheduler_addr, port, storage_url)) {
```

Comment:
```suggestion
    if (false == parse_args(argc, argv, scheduler_addr, port, storage_url)) {
```

To follow our contribution guidelines.